### PR TITLE
!refactor: implement staking MsgServer instead of multistaking MsgServer

### PR DESCRIPTION
**Pull request by GNaD13**
# Description
- As we remove  `intermediary delegator`, there's no need for a clear separation of `multistaking msg` (by multistaker) -> `staking msg` (by intermediry delegator). User now directly manage the underlying delegation so it makes more sense to just use staking txs instead of multi-staking txs.
- Also, we can remove the multi-staking txs and use staking txs since these two have the same fields. Doing so allows for UI compatibility with UI like keplr.

# Changes in this PR
- Remove msg types of multi-staking and related logic.
- Implement staking MsgServer

### CODE REVIEW

File: x/multi-staking/keeper/keeper.go [hunk 1]
```diff
@@ -39,14 +38,13 @@ func NewKeeper(
...
-		stakingMsgServer: stakingkeeper.NewMsgServerImpl(stakingKeeper),
-		distrMsgServer:   distrkeeper.NewMsgServerImpl(distrKeeper),
-		govMsgServer:     govkeeper.NewMsgServerImpl(govKeeper),
-		bankKeeper:       bankKeeper,
+		cdc:            cdc,
+		storeKey:       key,
+		memKey:         memKey,
+		stakingKeeper:  stakingKeeper,
+		distrMsgServer: distrkeeper.NewMsgServerImpl(distrKeeper),
+		govMsgServer:   govkeeper.NewMsgServerImpl(govKeeper),
```
Comment (catShaark) [inline x/multi-staking/keeper/keeper.go:46]: we should remove this unused `govkeeper`.

File: x/multi-staking/keeper/keeper.go [hunk 2]
```diff
@@ -19,14 +19,13 @@ import (
...
-	stakingKeeper    types.StakingKeeper
-	stakingMsgServer stakingtypes.MsgServer
-	distrMsgServer   distrtypes.MsgServer
-	govMsgServer     govtypes.MsgServer
-	bankKeeper       types.BankKeeper
+	storeKey       storetypes.StoreKey
+	memKey         storetypes.StoreKey
+	cdc            codec.BinaryCodec
+	stakingKeeper  stakingkeeper.Keeper
+	distrMsgServer distrtypes.MsgServer
```
Comment (catShaark) [inline x/multi-staking/keeper/keeper.go:26]: we should remove distrMsgServer and govMsgServer

**Comment by catShaark** [comment]
Can you remove all the unused code such as tx.pb.go or the related register logic.

**Comment by GNaD13** [comment]
> Can you remove all the unused code such as tx.pb.go or the related register logic.

removed

**Pull request merged**